mathlibtools/lib.py

```diff
@@ -1012,3 +1010,27 @@ def pull(self, remote: str='origin') -> None:
                 self.run_echo(['leanpkg', 'configure'])
             self.get_mathlib_olean()
 
+    def port_status(self, url: Optional[str] = None) -> None:
+        """
+        Color nodes on the graph based on port status. Done in place to the graph nodes.
+
+        Args:
+            url: md or yaml file with "file: label" content, by default, from the wiki
+        """
+        if url is None:
+            url = 'https://raw.githubusercontent.com/wiki/leanprover-community/mathlib/mathlib4-port-status.md'
+        def yaml_md_load(wikicontent: bytes):
+            return yaml.safe_load(wikicontent.replace(b"```", b""))
+
+        port_labels: Dict[str, str] = yaml_md_load(requests.get(url).content)
+
+        for filename, status in port_labels.items():
+            if filename not in self.import_graph.nodes:
+                continue
+            node = self.import_graph.nodes[filename]
+            if 'yes' in status.lower():
+                node["fillcolor"] = "green"
+                node["style"] = "filled"
+            elif 'no' in status.lower():
+                node["fillcolor"] = "red"
+                node["style"] = "filled"
```

Comment:
Could we please have things that just say `No` not be colored at all?

But things that say `No` followed by one or more characters to be shown in yellow or orange?

Maybe we could use `PaleGreen` for yes, and `LightSalmon` for no-with-extra-info?

Reply:
[algebra.order.ring.pdf](https://github.com/leanprover-community/mathlib-tools/files/9825316/algebra.order.ring.pdf)

Reply:
I agree about no color on plain Nos. How about this?

Reply:
Actually, how about we are more specific about the `No`. Having a visual distinction between 'No: blocked' and 'No: WIP' is helpful. If it contains `WIP` or `PR`, than shade it something nice (maybe that same light blue?), and if it contains No and more characters, but not WIP or PR, then shade it orange.